### DISCUSSION TITLE
bug or feature? deleted external library - assets still shown

### DISCUSSION THREAD

**Discussion by aisbergde** [immich-app/immich#7150] [Q&A] [answered]
I deleted an external library in the user settings. But this has no effect, all assets are still shown.

- what exactly happens, when an external library is deleted via GUI?
- in the table `libraries` it has an entry in `deletedAt`  -> fine
- `select count(*) from assets` -> all assets of the deleted library are still in the database -> could be fine, but there should be possible some kind of maintenance.
- Shouldn't it be the case that all assets in a deleted library are no longer displayed?
- What is necessary to finally delete the assets of the deleted external library from the database? I don't mean how I could do this via a database query. But how can I be prompted via the GUI to remove the deleted assets as well?

**Comment by schuhbacca** [comment, marked as answer]
I assume it will be totally removed when the nightly job is run to refresh all libraries. You can try and hit "Scan All Libraries" in the meantime to see if that does the trick.

**Reply by aisbergde** [reply]
perfect answer!

**Reply by delirious11** [reply]
did this work for you?   I am trying to redo an external library and the assets are still there after deleting the library.  May take some time, so I thought I'd ask

**Reply by aisbergde** [reply]
it worked. I manually started the rescan and assets of the deleted library have been removed.

**Reply by delirious11** [reply]
In my case,  I had multiple thumbnails and decided to just start over and remove the volumes from portainer and start over.  This was much faster.   https://immich.app/docs/FAQ/#how-can-i-purge-data-from-immich    I assume, since I started and stopped my upload so many times using different installations, the volumes were just wrong.   This allowed me a clean start.  Started the external library last night and it's looking good this morning.

**Reply by Gitonthegrid** [reply]
How do you force the rescan after you removed the external library?
It is really a anoying bug when you want to try some stuff in immich. 

Ik keeps showing all the pictures, even though I removed the complete external library. 
When the library was still active, I was trying the "remove offline files" but also that didnt work at all. 
I was hoping with removing the library, it would immediately resan everything and remove them from the "photos"

**Reply by aisbergde** [reply]
Try this
![image](https://github.com/immich-app/immich/assets/9020187/00adc7fc-f2e3-41be-8c05-95af75d8ead5)

**Reply by Gitonthegrid** [reply]
Already tried that a couple of times
